external/src/main/java/com/redhat/cloud/policies/engine/LightweightEngineImpl.java

```diff
@@ -0,0 +1,265 @@
+
+    @Inject
+    /*
+     * This channel is already used in NotificationActionPluginListener. Because of that, the
+     * mp.messaging.outgoing.webhook.merge config key is required in applications.properties.
+     */
+    @Channel(WEBHOOK_CHANNEL)
+    @OnOverflow(UNBOUNDED_BUFFER)
+    Emitter<String> emitter;
+
+    @Inject
+    PoliciesHistoryService policiesHistoryService;
+
+    @PostConstruct
+    void postConstruct() {
+        boolean enabled = ConfigProvider.getConfig().getOptionalValue(LIGHTWEIGHT_ENGINE_CONFIG_KEY, Boolean.class).orElse(false);
+        LOGGER.infof("Lightweight engine is %s", enabled ? "enabled" : "disabled");
+    }
+
+    @Override
+    public void loadTrigger(FullTrigger fullTrigger) {
+        LOGGER.debugf("Loading trigger %s", fullTrigger);
+        provideDefaultDampening(fullTrigger);
+        String tenantId = fullTrigger.getTrigger().getTenantId();
+        triggersByTenant.computeIfAbsent(tenantId, unused -> new HashMap<>()).put(fullTrigger.getTrigger().getId(), fullTrigger);
+    }
+
+    @Override
+    public void reloadTrigger(Trigger trigger, Collection<Condition> conditions) {
+        LOGGER.debugf("Reloading trigger %s with conditions %s", trigger, conditions);
+        boolean found = false;
+        Map<String, FullTrigger> tenantTriggers = triggersByTenant.get(trigger.getTenantId());
+        if (tenantTriggers != null) {
+            FullTrigger fullTrigger = tenantTriggers.get(trigger.getId());
+            if (fullTrigger != null) {
+                fullTrigger.setTrigger(trigger);
+                fullTrigger.setConditions(new ArrayList<>(conditions));
+                found = true;
+                LOGGER.debugf("Trigger reloaded");
+            }
+        }
+        if (!found) {
+            LOGGER.debugf("Trigger to reload not found");
+        }
+    }
+
+    @Override
+    public void removeTrigger(String tenantId, String triggerId) {
+        LOGGER.debugf("Removing trigger %s/%s", tenantId, triggerId);
+        boolean found = false;
+        Map<String, FullTrigger> tenantTriggers = triggersByTenant.get(tenantId);
+        if (tenantTriggers != null) {
+            found = tenantTriggers.remove(triggerId) != null;
+            LOGGER.debugf("Trigger removed");
+        }
+        if (!found) {
+            LOGGER.debugf("Trigger to remove not found");
+        }
+    }
+
+    @Override
+    public void process(Event event) {
+        LOGGER.debugf("Processing event %s", event);
+        Set<Alert> alerts = fireTriggers(event);
+        if (alerts.isEmpty()) {
+            LOGGER.debug("No alerts created from the event");
+        } else {
+            LOGGER.debugf("%d alerts created from the event", alerts.size());
+
+            // In the old implementation, the time comes from the Action constructor which relies on System.currentTimeMillis().
+            LocalDateTime nowUTC = LocalDateTime.ofInstant(Instant.ofEpochMilli(System.currentTimeMillis()), UTC);
```

Comment:
I guess it would be OK to use `LocalDateTime.now`, would that be a way different value? We can start with this currentTimeMillis anyhow.

Reply:
We could use it, yes. I purposely changed as few code as possible in this PR to avoid introducing bugs and behavior differences though, so let's keep that change for another PR if that's fine with you :smiley:

Reply:
It's fine by me.